### PR TITLE
Restore WeViewProxyView.

### PR DESCRIPTION
**Pull request by charlesmchen**
As discussed yesterday, I think this offers a more straightforward solution for your layout needs.  This class had previously existed so I resurrected it.

PTAL @bridger 
CC @DawnWright

### CODE REVIEW

**Comment by bridger** [comment]
This would be perfect. LGTM

File: WeView/WeViewProxyView.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//
+//  WeView.h
```
Comment (DawnWright) [inline WeView/WeViewProxyView.h]: Comment here has wrong filename and copyright date.

File: WeView/WeViewProxyView.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// An invisible view whose desired size is based on another view.  The other view IS NOT a subview
+// of the proxy.
+//
+// This proxy is useful for balancing layouts, ie. by having an invisible view that counterbalances
+// the space used in the layout by another view.
+@interface WeViewProxyView : UIView
+
+// A factory method that returns an instance of WeViewProxyView that maintains a strong reference to
+// the other view.
++ (WeViewProxyView*)proxyWithView:(UIView *)view;
```
Comment (DawnWright) [inline WeView/WeViewProxyView.h]: ws here seems different than the rest of weview (missing space before *)

File: WeView/WeViewProxyView.h (hunk 1 above)
Comment (charlesmchen) [inline WeView/WeViewProxyView.h]: Done in #3. Yes, precommit won't work with WeView (or other third party libs).

File: WeView/WeViewProxyView.h (hunk 2 above)
Comment (charlesmchen) [inline WeView/WeViewProxyView.h]: Done in #3.

**Pull request merged**